packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -79,57 +109,141 @@ public function getAliasName(int $domainId, string $index): string
      */
     public function createIndex(int $domainId, string $index)
     {
-        $definition = $this->getDefinition($domainId, $index);
+        $definition = $this->getStructureDefinition($domainId, $index);
         $indexes = $this->client->indices();
-        $aliasName = $this->getAliasName($domainId, $index);
-        $indexName = $this->getIndexName($domainId, $index);
+        $indexName = $this->getVersionedIndexName($domainId, $index);
         if ($indexes->exists(['index' => $indexName])) {
+            if ($this->buildVersion === 0) {
+                throw new ElasticsearchStructureException('Index %s already exists. Please start using build version or delete current index before creating a new one.');
+            }
             throw new ElasticsearchStructureException(sprintf('Index %s already exists', $indexName));
```

Comment:
Maybe when the buildVersion is in use and the index already exists, we can tell the user to run `php phing generate-build-version` as that seems like a thing people can miss. I've forgot to do that while testing this earlier:
```php
} else {
    $message .= ' Maybe you forgot to execute "php phing generate-build-version" before creating new index?';
}
throw new ElasticsearchStructureException($message);
```